### PR TITLE
fix(runtime): Pages Router links from localized to base pages don't work

### PR DESCRIPTION
**Pull request by agurtovoy**
## Before

Dev: 

https://github.com/user-attachments/assets/c3721867-c1b8-472a-9fc8-f39ae545e558

Prod:

https://github.com/user-attachments/assets/9b02179b-d4d7-4b66-b446-34ff1420bf99

## After

Dev:

https://github.com/user-attachments/assets/67cac23c-a773-43b7-82ca-f74f412a4562

Prod:

https://github.com/user-attachments/assets/09aa264c-13b6-4d5a-aad9-0a94e044766e

### CODE REVIEW

File: packages/runtime/src/components/shared/Link/index.tsx [hunk 1]
```diff
@@ -147,9 +146,7 @@ export const Link = forwardRef<HTMLAnchorElement, Props>(function Link(
         target={target}
         onClick={handleClick}
         href={href}
-        {...(hasLocalizedPathname && {
-          locale: false,
-        })}
+        locale={false}
```
Comment (agurtovoy) [inline packages/runtime/src/components/shared/Link/index.tsx:149]: We don't need extra "help" from Next.js in handling localized links.

Comment (agurtovoy) [reply packages/runtime/src/components/shared/Link/index.tsx:149]: Note that this doesn't affect App Router (I triple checked), which doesn't offer built-in support for localization.

Comment (agurtovoy) [reply packages/runtime/src/components/shared/Link/index.tsx:149]: Also, as I mention in the [linked issue](https://linear.app/makeswift/issue/ENG-6510/pages-router-links-from-a-localized-page-to-base-pages-dont-work), links work as expected a Pages Router site is using domain-based localization (before and after the fix).

**Review by migueloller — APPROVED** [review]
Thanks for the extensive testing and thoroughness of the work here 🙏🏻

**Pull request merged**